From d664c4fedab549affd6fada0aadf7ffe45b9c82d Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 13 Jun 2021 23:42:19 -0500
Subject: [PATCH 043/120] hwspinlock: sun6i: Register layout is known

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 drivers/hwspinlock/sun6i_hwspinlock.c | 20 +++++++-------------
 1 file changed, 7 insertions(+), 13 deletions(-)

diff --git a/drivers/hwspinlock/sun6i_hwspinlock.c b/drivers/hwspinlock/sun6i_hwspinlock.c
index 28780be54e68..2e6c88eb9ba1 100644
--- a/drivers/hwspinlock/sun6i_hwspinlock.c
+++ b/drivers/hwspinlock/sun6i_hwspinlock.c
@@ -127,22 +127,16 @@ static int sun6i_hwspinlock_probe(struct platform_device *pdev)
 	}
 
 	/*
-	 * bit 28 and 29 represents the hwspinlock setup
-	 *
-	 * every datasheet (A64, A80, A83T, H3, H5, H6 ...) says the default value is 0x1 and 0x1
-	 * to 0x4 represent 32, 64, 128 and 256 locks
-	 * but later datasheets (H5, H6) say 00, 01, 10, 11 represent 32, 64, 128 and 256 locks,
-	 * but that would mean H5 and H6 have 64 locks, while their datasheets talk about 32 locks
-	 * all the time, not a single mentioning of 64 locks
-	 * the 0x4 value is also not representable by 2 bits alone, so some datasheets are not
-	 * correct
-	 * one thing have all in common, default value of the sysstatus register is 0x10000000,
-	 * which results in bit 28 being set
-	 * this is the reason 0x1 is considered being 32 locks and bit 30 is taken into account
-	 * verified on H2+ (datasheet 0x1 = 32 locks) and H5 (datasheet 01 = 64 locks)
+	 * Bits 28 and 29 represent the hwspinlock setup.
+	 * 0x0 == 0x4 -> 256 spinlocks.
+	 * All known hardware has 0x1 -> 32 spinlocks.
+	 * The H5 and H6 datasheets incorrectly list the values off by one.
 	 */
 	num_banks = readl(io_base + SPINLOCK_SYSSTATUS_REG) >> 28;
 	switch (num_banks) {
+	case 0:
+		num_banks = 4;
+		fallthrough;
 	case 1 ... 4:
 		priv->nlocks = 1 << (4 + num_banks);
 		break;
-- 
2.32.0

