From 05887bcdd339d4e7e4b3e0f82fe871304056fb20 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 13 Jun 2021 23:42:27 -0500
Subject: [PATCH 044/120] hwspinlock: sun6i: Fix debugfs leak

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 drivers/hwspinlock/sun6i_hwspinlock.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/hwspinlock/sun6i_hwspinlock.c b/drivers/hwspinlock/sun6i_hwspinlock.c
index 2e6c88eb9ba1..1af796a2e1a1 100644
--- a/drivers/hwspinlock/sun6i_hwspinlock.c
+++ b/drivers/hwspinlock/sun6i_hwspinlock.c
@@ -160,13 +160,11 @@ static int sun6i_hwspinlock_probe(struct platform_device *pdev)
 
 	/* failure of debugfs is considered non-fatal */
 	sun6i_hwspinlock_debugfs_init(priv);
-	if (IS_ERR(priv->debugfs))
-		priv->debugfs = NULL;
 
 	err = devm_add_action_or_reset(&pdev->dev, sun6i_hwspinlock_disable, priv);
 	if (err) {
 		dev_err(&pdev->dev, "failed to add hwspinlock disable action\n");
-		goto bank_fail;
+		goto action_fail;
 	}
 
 	platform_set_drvdata(pdev, priv);
@@ -174,6 +172,8 @@ static int sun6i_hwspinlock_probe(struct platform_device *pdev)
 	return devm_hwspin_lock_register(&pdev->dev, priv->bank, &sun6i_hwspinlock_ops,
 					 SPINLOCK_BASE_ID, priv->nlocks);
 
+action_fail:
+	debugfs_remove_recursive(priv->debugfs);
 bank_fail:
 	clk_disable_unprepare(priv->ahb_clk);
 clk_fail:
-- 
2.32.0

