From 4528ddffc52b50f0d9314fdedfcc0efd8265390c Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 13 Jun 2021 23:50:57 -0500
Subject: [PATCH 099/120] ASoC: sun4i-i2s: Also set capture DMA width

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 sound/soc/sunxi/sun4i-i2s.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/sunxi/sun4i-i2s.c b/sound/soc/sunxi/sun4i-i2s.c
index 98878b31de37..17797f196b7b 100644
--- a/sound/soc/sunxi/sun4i-i2s.c
+++ b/sound/soc/sunxi/sun4i-i2s.c
@@ -616,6 +616,7 @@ static int sun4i_i2s_hw_params(struct snd_pcm_substream *substream,
 			params_physical_width(params));
 		return -EINVAL;
 	}
+	i2s->capture_dma_data.addr_width = width;
 	i2s->playback_dma_data.addr_width = width;
 
 	sr = i2s->variant->get_sr(word_size);
-- 
2.32.0

