From f2afe044dc518adcc81bcfd927768ed8d022de46 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 23 Jun 2021 21:18:47 -0500
Subject: [PATCH 105/120] [WIP] ASoC: sun20i-codec: What is this ramp thing?

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 sound/soc/sunxi/sun20i-codec.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/sunxi/sun20i-codec.c b/sound/soc/sunxi/sun20i-codec.c
index f99e81f93a1b..18551f5ef14e 100644
--- a/sound/soc/sunxi/sun20i-codec.c
+++ b/sound/soc/sunxi/sun20i-codec.c
@@ -711,6 +711,10 @@ static int sun20i_codec_component_probe(struct snd_soc_component *component)
 				      0x3 << SUN20I_CODEC_ADC_DIG_CTRL_ADC_VOL_EN,
 				      0x3 << SUN20I_CODEC_ADC_DIG_CTRL_ADC_VOL_EN);
 
+	/* Maaagic... */
+	snd_soc_component_update_bits(component, SUN20I_CODEC_RAMP,
+				      BIT(1) | BIT(0), BIT(0));
+
 	return 0;
 
 err_assert_reset:
-- 
2.32.0

